TFNetworkLayer.py

```diff
@@ -855,31 +866,44 @@ def batch_norm(self, data,
         # Note: Our Theano implementation does not use a moving average for this.
         with self.var_creation_scope():
           sample_variance = self.add_param(tf.Variable(
-            initial_value=tf.ones(data.get_bc_spatial_batch_shape()),
+            initial_value=tf.ones(shape),
             name="%s_%s_variance" % (self.name, data.name),
             trainable=False))
         sample_variance = tf.assign_add(sample_variance, (variance - sample_variance) * momentum)
       # If train or if force_sample, use default use_sample=0.0, otherwise use_sample=1.0.
       use_sample = 1.0 + tf.cast(tf.logical_or(self.network.train_flag, force_sample), tf.float32) * (use_sample - 1.0)
       mean = (1. - use_sample) * mean + use_sample * sample_mean
       variance = (1. - use_sample) * variance + use_sample * sample_variance
-      bn = (data.placeholder - mean) * tf.rsqrt(variance + epsilon)
       if use_std:
         if gamma is None:
           with self.var_creation_scope():
+            if gamma_init == "zeros":
+              gamma_initializer = tf.zeros_initializer()
+            else:
```

Comment:
The `else:` is too generic. You should fail if there is an unexpected `gamma_init`.

Reply:
(But see my comment about `get_initializer`, then this is anyway obsolete.)